Add recipe for jira

unmonoqueteclea opened the pull request:
### Brief summary of what the package does

Emacs integration for [Altassian's Jira](https://www.atlassian.com/software/jira).

Supports listing and filtering issues, viewing issue details,
modifying certain properties, and adding worklogs.

Additionally, it provides support for displaying all worklogs from the
[Tempo](https://www.tempo.io/products/jira-time-tracking) integration.

### Direct link to the package repository

https://github.com/unmonoqueteclea/jira.el

### Your association with the package

I am the maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass -

Are you aware of https://github.com/emacsmirror/emacswiki.org/blob/master/jira.el ?  The Emacs wiki used to be (might still be?) an very informal package index where people housed elisp, and there already is something out there that's claimed the name `jira`.

⸺ `jira-api.el` with byte-compile using Emacs 30.1:
```
jira-api.el:96:2: Warning: docstring wider than 80 characters
```

⸺ `jira-utils.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- jira-utils.el#L173: Consider using %F instead of %Y-%m-%d in time strings
```

⸺ `jira-tempo.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- jira-tempo.el#L171: It's safer to sharp-quote function names; use `#'`
```

⸺ `jira-issues.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- jira-issues.el#L215: It's safer to sharp-quote function names; use `#'`
- jira-issues.el#L219: It's safer to sharp-quote function names; use `#'`
- jira-issues.el#L219: Loading a package should rarely add hooks
```

⸺ `jira-fmt.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- jira-fmt.el#L105: Consider using %F instead of %Y-%m-%d in time strings
```

⸺ `jira-detail.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- jira-detail.el#L73: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L75: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L78: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L80: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L82: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L85: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L87: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L89: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L101: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L102: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L104: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L105: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L115: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L116: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L117: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L118: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L119: `concat` may be unneeded; `insert` can take multiple arguments
- jira-detail.el#L120: `concat` may be unneeded; `insert` can take multiple arguments
```

⸺ `jira-actions.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- jira-actions.el#L73: Consider using %F instead of %Y-%m-%d in time strings
- jira-actions.el#L73: Consider using %T instead of %H:%M:%S in time strings
```

⸺ Package and license:
- Please specify `:fetcher` before `:repo` in your recipe
- Add a LICENSE file to the repository

unmonoqueteclea commented:
Thank you very much, @riscy. 
I've fixed all the warnings and integrated `melpazoid` in my Github repository so that I get warned automatically in future versions. I also added a `LICENSE` file.


> Are you aware of https://github.com/emacsmirror/emacswiki.org/blob/master/jira.el ? The Emacs wiki used to be (might still be?) an very informal package index where people housed elisp, and there already is something out there that's claimed the name `jira`.

Actually, no, I wasn’t aware of it. I searched the major package archives (ELPA, Non-GNU ELPA, and MELPA) but didn’t find anything. I checked the package you mentioned, and it relies on the XML-RPC JIRA API, which was removed in 2015. So, I don’t think anyone is still using it.

Do you still think I should rename my package? I’m new to the ecosystem as a package developer, so I’m happy to follow whatever you or any maintainer consider the right approach.

riscy commented:
> Do you still think I should rename my package?

I think in this case no, and the information that the emacswiki package wouldn't have worked since 2015 is helpful - thanks.

A couple last things:

- Please specify `:fetcher` before `:repo` in your recipe
- In case you haven't, ensure GitHub release v0.0.1 is up-to-date with your current code and `Package-Version`

unmonoqueteclea commented:
>     * Please specify `:fetcher` before `:repo` in your recipe

Done! Forgot to push it.

>     * In case you haven't, ensure GitHub release v0.0.1 is up-to-date with your current code and `Package-Version`

Yes, I checked it (also the package version in the module is "0.0.1" (without the prefix "v")

riscy commented:
Thank you!  I double-check the package-version thing with everyone *just in case*.

Welcome to MELPA! :)
